lms/services/roster.py

```diff
@@ -246,6 +248,78 @@ def fetch_assignment_roster(self, assignment: Assignment) -> None:
             update_columns=["active", "updated"],
         )
 
+    def fetch_canvas_group_roster(self, canvas_group: LMSSegment) -> None:
```

Comment:
> I would wait until a few more have been implemented, to make sure we don't create the wrong abstraction. Let's keep it as is for now.

Yes exactly. I reckon canvas_groups is going to be the most "different" it calls the name and roles service but the rest will call each own LMS API.

>At some point it might make sense to extract those to abstractions, and using the strategy pattern, make this service resolve the right one and call it.

Yeah that probably the right approach once we have a few of this implemented.